data/core/macros/utils.cfg

```diff
@@ -578,6 +578,12 @@
             name=attack end
             delayed_variable_substitution=yes
 
+            [filter_condition]  
+                [have_unit]
+                    id=$unit.id
+                [/have_unit]
+            [/filter_condition]
+
```

Comment:
I don't see how this condition can ever be false. `attack end` fires before `die` so both $unit and $second_unit will always exist here, won't they?

Reply:
If an event where unit or second_unit disapear before attack end was called then condition is false. But not example in core, but in an addon campaign of @shikadiqueen. The bug was teported in forum of After_the_storm campaign page 33 by white_haired_uncle but i was confronted in same problems on windows 7.